### PR TITLE
Fix an expand issue with bool

### PR DESCRIPTION
**Pull request by chinhuang007**
If the input data type is bool, the expand operator would
fail to convert and throw an exception, as described in
https://github.com/onnx/onnx-tensorflow/issues/600

This PR converts input data to tf.int32 for matmul when
dtype is bool.

### CODE REVIEW

**Comment by chinhuang007** [comment]
@tjingrant @fumihwh Could you please review this simple fix to resolve an open user issue?

**Comment by tjingrant** [comment]
Hi Chin, how does this approach compare with, say using `tf.slice`? It seems like you have to materialize the coordinate vector for each pixel that is not removed by negative padding, and using `tf.gather` also carries performance penalties, given that it's fundamentally a sparse operation.

File: onnx_tf/handlers/backend/expand.py [hunk 1]
```diff
@@ -11,5 +11,12 @@ class Expand(BackendHandler):
   def version_8(cls, node, **kwargs):
     tensor_dict = kwargs["tensor_dict"]
     x, shape = tensor_dict[node.inputs[0]], tensor_dict[node.inputs[1]]
-    ones = tf.ones(shape, dtype=x.dtype)
-    return [x * ones]
+
+    # tf.matmul does not support bool therefore use int32
```
Comment (tjingrant) [inline onnx_tf/handlers/backend/expand.py:15]: Why is tf.matmul relevant here?

Comment (chinhuang007) [reply onnx_tf/handlers/backend/expand.py]: Yeah, wrong comment, it is tf.math.multiply, not matmul.

**Comment by tjingrant** [comment]
Also just minor suggestion, int32 expands the memory footprint by at least 4x. Using int8 may be more appropriate if it's possible & easy to do so.

**Comment by chinhuang007** [comment]
Makes sense to use smaller int8. Thanks for the suggestion.

**Pull request merged**